### PR TITLE
introspection: implement prometheus http

### PR DESCRIPTION
**Pull request by ldelossa**
this commit implements promethemeus metrics and removes OTEL usage for
metrics.

dogstatd metrics are now deprecated infavor of conforming on prometheus
metrics.

Signed-off-by: ldelossa <ldelossa@redhat.com>

### CODE REVIEW

File: config/introspection.go [hunk 1]
```diff
@@ -20,16 +22,15 @@ type Jaeger struct {
...
 type Metrics struct {
 	Name       string     `yaml:"name" json:"name"`
 	Prometheus Prometheus `yaml:"prometheus" json:"prometheus"`
-	Dogstatsd  Dogstatsd  `yaml:"dogstatsd" json:"dogstatsd"`
 }
 
+// Prometheus specific metrics configuration
 type Prometheus struct {
+	// an path endpoint parsable by url.Parse where
+	// prometheus metrics will be scraped.
```
Comment (hdonnay) [inline config/introspection.go]:
```suggestion
	// Endpoint is a URL path where
	// Prometheus metrics will be hosted.
```

File: httptransport/server.go [hunk 2]
```diff
@@ -168,52 +161,20 @@ func (t *Server) configureComboMode(ctx context.Context) error {
...
-	)
-	t.Handle(IndexStateAPIPath, othttp.WithRouteTag(IndexStateAPIPath, stateH))
+	t.Handle(AffectedManifestAPIPath,
+		intromw.InstrumentedHandler(AffectedManifestAPIPath, t.traceOpt, AffectedManifestHandler(t.indexer)))
+
+	t.Handle(IndexAPIPath,
+		intromw.InstrumentedHandler(IndexAPIPath, t.traceOpt, IndexHandler(t.indexer)))
+
+	t.Handle(IndexReportAPIPath,
+		intromw.InstrumentedHandler(IndexReportAPIPath+"GET", t.traceOpt, IndexReportHandler(t.indexer)))
```
Comment (hdonnay) [inline httptransport/server.go:174]: Why does this have an extra string tacked on?

File: middleware/introspection/instrumentedhandler.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	h = promhttp.InstrumentHandlerRequestSize(RequestSize, h)
+	h = promhttp.InstrumentHandlerResponseSize(ResponseSize, h)
+	h = promhttp.InstrumentHandlerTimeToWriteHeader(RequestDuration, h)
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		recorder := rr.NewResponseRecorder(w)
+		h.ServeHTTP(w, r)
+		zlog.Info(r.Context()).
+			Str("remote addr", r.RemoteAddr).
+			Str("method", r.Method).
+			Str("request uri", r.RequestURI).
```
Comment (hdonnay) [inline middleware/introspection/instrumentedhandler.go]:
```suggestion
			Str("remote_addr", r.RemoteAddr).
			Str("method", r.Method).
			Str("request_uri", r.RequestURI).
```

**Review by hdonnay — CHANGES_REQUESTED** [review]

File: httptransport/server.go (hunk 2 above)
Comment (ldelossa) [reply httptransport/server.go:174]:
```
	IndexAPIPath            = indexerRoot + apiRoot + "index_report"
	IndexReportAPIPath      = indexerRoot + apiRoot + "index_report/"
```

when both those paths are normalized to a Prom metric, if we don't tack on an extra string to differentiate, they will have the same metric name, and registration of the metric will panic.

**Review by hdonnay — DISMISSED** [review]
LGTM

**Review by hdonnay — APPROVED** [review]

**Pull request merged**